libsoftwarecontainer/src/container.cpp

```diff
@@ -601,7 +605,31 @@ ReturnCode Container::bindMountFolderInContainer(const std::string &pathOnHost,
     }
 
     result = gatewaysDirInContainer() + "/" + pathInContainer;
-    return ReturnCode::SUCCESS;
+
+    if (readonly) {
+        return remountReadOnlyInContainer(result);
+    } else {
+        return ReturnCode::SUCCESS;
+    }
+}
+
+ReturnCode Container::remountReadOnlyInContainer(const std::string &path)
+{
+    pid_t pid = INVALID_PID;
+    EnvironmentVariables envVars;
```

Comment:
The env var 'variables' parameter in executeInContainer has a default value so if we don't plan to pass anything here I guess we can remove the env vars from this part of the code.